tests/components/mqtt/test_init.py

```diff
@@ -1063,6 +1117,121 @@ async def test_setup_logs_error_if_no_connect_broker(hass, caplog):
+        )
+    await hass.async_block_till_done()
+    assert "No ACK from MQTT server in 10 seconds" in caplog.text
+
+
+async def test_publish_error(hass, caplog):
+    """Test publish error."""
+    entry = MockConfigEntry(domain=mqtt.DOMAIN, data={mqtt.CONF_BROKER: "test-broker"})
+
+    # simulate an Out of memory error
+    with patch("paho.mqtt.client.Client") as mock_client:
+        mock_client().connect = lambda *args: 1
+        mock_client().publish().rc = 1
+        assert await mqtt.async_setup_entry(hass, entry)
+        await hass.async_block_till_done()
+        with pytest.raises(HomeAssistantError):
+            await mqtt.async_publish(
+                hass, "some-topic", b"test-payload", qos=0, retain=False, encoding=None
+            )
+        assert "Failed to connect to MQTT server: Out of memory." in caplog.text
+
+
+async def test_handle_message_callback(hass, caplog):
+    """Test for handling an incoming message callback."""
+    entry = MockConfigEntry(domain=mqtt.DOMAIN, data={mqtt.CONF_BROKER: "test-broker"})
+    msg = ReceiveMessage("some-topic", b"test-payload", 0, False)
+    with patch("paho.mqtt.client.Client") as mock_client:
+        mock_client().connect = lambda *args: 0
+        assert await mqtt.async_setup_entry(hass, entry)
+        await hass.async_block_till_done()
+        await mqtt.async_subscribe(hass, "some-topic", lambda *args: 0)
+        component = hass.data["mqtt"]
+        component._mqtt_on_message(mock_client, None, msg)
+
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    assert "Received message on some-topic: b'test-payload'" in caplog.text
+
+
+async def test_setup_override_configuration(hass, caplog, tmp_path):
+    """Test override setup from configuration entry."""
+    # Mock password setup from config
+    config = {
+        "password": "someyamlconfiguredpassword",
+    }
+    new_yaml_config_file = tmp_path / "configuration.yaml"
+    new_yaml_config = yaml.dump({mqtt.DOMAIN: config})
+    new_yaml_config_file.write_text(new_yaml_config)
+    assert new_yaml_config_file.read_text() == new_yaml_config
+
+    with patch.object(hass_config, "YAML_CONFIG_FILE", new_yaml_config_file):
+        # Mock config entry
+        entry = MockConfigEntry(
+            domain=mqtt.DOMAIN,
+            data={mqtt.CONF_BROKER: "test-broker", "password": "somepassword"},
+        )
+
+        with patch("paho.mqtt.client.Client") as mock_client:
+            mock_client().connect = lambda *args: 1
+            await async_setup_component(hass, mqtt.DOMAIN, {mqtt.DOMAIN: config})
+            await entry.async_setup(hass)
+            await hass.async_block_till_done()
+
+            assert (
+                "Data in your configuration entry is going to override your configuration.yaml:"
+                in caplog.text
+            )
+
+
+async def test_setup_mqtt_client_protocol(hass):
```

Comment:
This test doesn't seem to verify the protocol is passed to the PAHO client?

Reply:
done